pennylane/qchem/hartree_fock.py

```diff
@@ -125,17 +125,18 @@ def _scf(*args):
         charges = mol.nuclear_charges
         r = mol.coordinates
         n_electron = mol.n_electrons
+        argnums = [mol.coord_opt, mol.coeff_opt, mol.alpha_opt]
 
-        if r.requires_grad:
+        if getattr(r, "requires_grad", False) or argnums[0]:
```

Comment:
argnums ordering by default is assumed to be [coord, coeff, alpha].
argnums[0] always refers to the coordinates.

Reply:
It may also be necessary to need an extra check to ensure that they're using a JAX array or an array that is actually differentiable?